File: lib/java/macos/src/main/java/alphaTab/alphaSkia/AlphaSkiaMacOs.java
package alphaTab.alphaSkia;

/**
 * The alphaSkia initializer for macOS.
 */
public class AlphaSkiaMacOs extends AlphaSkiaJre {
    /**
     * The alphaSkia initializer for macOS.
     */
    public static final AlphaSkiaMacOs INSTANCE = new AlphaSkiaMacOs();

    private AlphaSkiaMacOs() {
    }

    @Override
    protected String[] getJavaResources() {
        return new String[] {
            "/native/macos-" + getCurrentArchitecture() + "/libalphaskiajni.dylib"
        };
    }
}

File: lib/kotlin/macos/src/main/java/alphaTab/alphaSkia/AlphaSkiaMacOs.kt
package alphaTab.alphaSkia

/**
 * The alphaSkia initializer for macOS.
 */
class AlphaSkiaMacOs private constructor() : AlphaSkiaJre() {
    override fun getJavaResources(): Array<String> {
        return arrayOf(
            "/native/macos-" + getCurrentArchitecture() + "/libalphaskiajni.dylib"
        )
    }

    companion object {
        /**
         * The alphaSkia initializer for Linux.
         */
        @JvmStatic
        val INSTANCE = AlphaSkiaMacOs()
    }
}

